sklearn/covariance/outlier_detection.py

```diff
@@ -62,9 +74,18 @@ def decision_function(self, X, raw_values=False):
             such as the One-Class SVM.
 
         """
+
+        if raw_values is not None:
+            warnings.warn(
+                'raw_values keyword has been moved to class initialization',
+                DeprecationWarning)
+
         check_is_fitted(self, 'threshold_')
         mahal_dist = self.mahalanobis(X)
-        if raw_values:
+        if self.raw_decision:
+            decision = -mahal_dist
+        elif raw_values is not None:
```

Comment:
it should be ``elif raw_values``, right? That was the check before. So if someone passed, say ``False``, they should not get the raw values.